src/main/pluginsServer.ts

```diff
@@ -141,23 +141,29 @@ export async function startPluginsServer(
             void jobQueueConsumer?.resume()
         })
 
-        // use one extra connection for redis pubsub
-        pubSub = await createRedis(hub)
-        await pubSub.subscribe(hub.PLUGINS_RELOAD_PUBSUB_CHANNEL)
-        pubSub.on('message', async (channel: string, message) => {
-            if (channel === hub!.PLUGINS_RELOAD_PUBSUB_CHANNEL) {
+        // use one extra Redis connection for pub-sub
+        pubSub = new PubSub(hub, {
+            [hub.PLUGINS_RELOAD_PUBSUB_CHANNEL]: async () => {
                 status.info('⚡', 'Reloading plugins!')
-
                 await piscina?.broadcastTask({ task: 'reloadPlugins' })
                 await scheduleControl?.reloadSchedule()
-            }
+            },
+            'reload-action': async (message) =>
+                await piscina?.broadcastTask({ task: 'reloadAction', args: { actionId: parseInt(message) } }),
+            'drop-action': async (message) =>
+                await piscina?.broadcastTask({ task: 'dropAction', args: { actionId: parseInt(message) } }),
         })
+        await pubSub.start()
 
         if (hub.jobQueueManager) {
             const queueString = hub.jobQueueManager.getJobQueueTypesAsString()
             await hub!.db!.redisSet('@posthog-plugin-server/enabled-job-queues', queueString)
         }
 
+        // every 5 minutes all ActionManager caches are reloaded for eventual consistency
+        pingJob = schedule.scheduleJob('*/5 * * * *', async () => {
```

Comment:
think this is overridden by the `pingJob` below?

Reply:
Good catch